src/main/java/pl/wip/wipplatformbe/validation/EmailValidator.java

```diff
@@ -0,0 +1,24 @@
+package pl.wip.wipplatformbe.validation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import org.springframework.beans.factory.annotation.Autowired;
+import pl.wip.wipplatformbe.services.UserService;
+
+public class EmailValidator implements ConstraintValidator<UniqueEmail, String> {
+
+    @Autowired
```

Comment:
Wiring fields using @Autowired is a common practice in Spring applications, including Spring Boot. However, it's generally considered less preferable compared to constructor injection, especially when there's only one constructor. (Read more about constructor injections)

When you use field injection (@Autowired on fields), Spring directly injects dependencies into the fields of your class. While this approach is convenient, it has some drawbacks:
1. Hidden Dependencies: With field injection, dependencies are not explicitly declared in the constructor, which can make it less clear what dependencies a class 
2. Testing Difficulty: Field injection can make unit testing more challenging because you cannot directly pass dependencies into the constructor when creating instances of the class for testing. This can lead to more complex setup procedures in tests.
3. Limited Flexibility: Field injection ties the lifecycle of the dependency to the lifecycle of the bean. This can limit the flexibility of managing dependencies, especially when you need different instances of the same dependency for different beans.

Creating this app we're trying to learn new things. Using Autowired on fields makes code less readable and harder to understand for beginners as they might not know what DI is.